client-core/src/synchronizer.rs

```diff
@@ -19,6 +20,9 @@ type WalletPassphrase = SecKey;
 pub struct PollingSynchronizer {
     wallets: Arc<Mutex<BTreeMap<WalletName, WalletPassphrase>>>,
     progress: Arc<Mutex<SynchronizerProgress>>,
+    should_run: Arc<AtomicBool>,
```

Comment:
houw about just using bool?

Reply:
It’s shared between threads, so atomic is needed.